src/main/resources/avro/genotypephenotypemethods.avdl

```diff
@@ -0,0 +1,144 @@
+
+Evidence:
+Evidence Code Ontology (ECO): http://www.ontobee.org/browser/index.php?o=ECO
+Ontology for Biomedical Investigations (OBI): http://purl.obofoundry.org/obo/obi/browse
+*/
+protocol GenotypePhenotypeMethods {
+
+
+import idl "genotypephenotype.avdl";
+import idl "methods.avdl";
+import idl "common.avdl";
+
+/******************  /genotypephenotype/search  *********************/
+
+
+
+/**
+Evidence for the phenotype association.
+*/
+record EvidenceQuery {
+  /** ECO or OBI is recommended */
+  array<org.ga4gh.models.OntologyTerm> evidenceType;
+}
+
+
+/**
+The feature collection to search for.  One or more features (variants,
+genes, etc) can be specified.  The idea here is that if a
+query is for a gene, then any alleles to that gene for which
+there are annotation records would be returned.
+*/
+record GenomicFeatureQuery {
+  array<org.ga4gh.models.Feature> features;
+}
+
+
+/**
+One or more phenotypes can be queried together.
+*/
+record PhenotypeQuery {
+  array<org.ga4gh.models.PhenotypeInstance> phenotypes;
+}
+
+/**
+One or more ontology terms can be queried together.
+*/
+record OntologyTermQuery {
+  array<org.ga4gh.models.OntologyTerm> terms;
+}
+
+/**
+One or more ids can be queried together.  Generally used for instances
+of a particular class of object (e.g. a specific gene or SNP).
+*/
+record ExternalIdentifierQuery {
+  array<org.ga4gh.models.ExternalIdentifier> ids;
+}
+
+/**
+This request maps to the body of `POST /genotypephenotype/search` as JSON.
+
+The goal here is to allow users to query using one or more of 
+Genotype, Phenotype, Environment, and Evidence.
+
+A query using one of the above items is to mean, by default,
+that the remainder of the query is as a "wildcard", such
+that all matches to just that query term would come back.
+Combinations of the above are to act like AND rather than OR.
+
+The "genotype" part of the query methods can be one or more
+genomic features.  Associations can be made at many
+levels of granularity (from whole genotypes down to individual
+SNVs), but users may use these methods with partial or
+inexact information.  Therefore, the query methods must be
+able to support query of some or all of the associated features.
+Furthermore, use of the relationships between genomic features
+means that when querying for a gene, any variants to that
+gene are also returned.  For example, a query with
+BRCA2 would mean that in addition to any direct associations
+to the BRCA2, all associations to sequence variants of BRCA2 would also
+be returned.  Similarly, queries with OntologyTerms should perform
+the subclass closure.
+
+Each query can be made against a string, an array of external 
+identifers (such as for gene or SNP ids), ontology term ids, or 
+full feature/phenotype/evidence objects.
+*/
+record SearchFeaturesRequest {
+
+  union {null, string, ExternalIdentifierQuery, OntologyTermQuery, 
+         GenomicFeatureQuery} feature = null;
```

Comment:
an alternative is to have distinct queries depending on the object type being supplied in the query.  this would be the least ambiguous. 
so instead of 
<pre>
SearchFeaturesRequest {
  union {null, string, ExternalIdentifierQuery, OntologyTermQuery, 
         GenomicFeatureQuery} feature = null;
} 
</pre>

we might proliferate the searches to:
<pre>
SearchGenomicFeatureByIDRequest {
  ExternalIdentifierQuery feature;
} 
SearchGenomicFeatureByOntologyTermRequest {
  OntologyTermQuery feature;
} 
SearchGenomicFeatureByStringRequest {
  string feature;
} 
SearchGenomicFeatureByGenomicFeatureRequest {
  GenomicFeatureQuery feature;
} 
</pre>

this would be the cleanest implementation.  we would leave these as their ```*FeatureQuery``` record types to allow extension in the future (for example, new parameters specific for the object type).